### DISCUSSION TITLE
Asset accounts current balance does not match with the banks current balance

### DISCUSSION THREAD

**Discussion by joerg-wille** [firefly-iii/firefly-iii#8751] [Firefly III]
I am using the Firefly in a docker container and [firefly-iii-fints-importer](https://github.com/bnw/firefly-iii-fints-importer) to import from my bank.(Both are latest version.) I have 3 asset accounts (2 default and 1 savings account) and there are occasionally transfers between these accounts. I mention this, because after importing these transfers are sometimes being imported twice, once as transfer and once as expense/revenue. I corrected these by deleting the duplicate expense/revenue entry.
But with or without my manual correction the "current balance" is completely different then the reality shown in my online banking.
I am quite new to Firefly - been using it now for 4 month - and do not have any/much knowledge in banking stuff - I am software developer ;-)
In the last 4 month I started 3 time from scratch, deleted db and all persisted data and imported last 4 years of data from my bank (using a correct start amount). At that time the current balance shown matched with my bank. Occasionally I import new data from my bank. The reason why I keep starting from scratch is that after 2 to 4 imports the current balance does not match any more.
Today I looked over all the imported data, each entry is correct (apart from 2 duplicated transfers). I have now idea where the difference in the current value originates from - as it does not seem to have anything todo with duplicated transfers.
What am I doing wrong?

A second problem I am facing. When I remove all imported transactions which resulted in the wrong current balance, I do not get back the balance as before the import. That's why I help myself with starting from scratch.

**Comment by JC5** [comment]
1. What happens when you browse to `/flush` after you remove stuff?
2. Did you consider using the reconciliation function? It's an option on the `/accounts/asset` page in the sub-menu for each account on the right. Tick off each transaction, and you'll spot the difference quickly enough.

When that doesn't help, use the "audit report" under `/reports` to see when the difference starts to show.

**Reply by joerg-wille** [reply]
Thanks for quick answer.

1. I have not used `/flush` when I was trying to correct the "current balance". I can try that next time when I have duplicates. But I have used `/flush` after I deleted imported transactions before trying to re-import these again. But flush and also restating all 3  firefly containers does not help. When I try to re-import, I get an error from firefly3-fints-importer that all transactions a "Duplicate of transaction" and "These transactions were most likely not imported."
2. I am aware of the reconciliation function, but I do not understand it's use-case other than correcting a wrong start offset - which is not my problem.
3. Thank you very much for the tip with the "audit report" - this report can really help me, when I am facing a wrong balance next time. (First I am trying to figure out, if I can repair the existing account or if I once more need to start from scratch. Is it possible to really delete the transactions so a re-import works?)

**Reply by JC5** [reply]
In regards to the first point, please read the documentation: https://docs.firefly-iii.org/how-to/data-importer/import/duplicates/
